### PR TITLE
Simulation camera

### PR DESCRIPTION
**Pull request by garygra**
Camera simulation working on macos. Need to check on linux.
`roslaunch mujoco_ros mushr.launch` <- Launch simulation with the camera on top of markers
`rqt_image_view` <- to check that the image is being published

### CODE REVIEW

File: infrastructure/prx_models/models/aruco_mip_36h12/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+
+execute_process(
```
Comment (aravindsiv) [inline infrastructure/prx_models/models/aruco_mip_36h12/CMakeLists.txt:3]: Would it be possible to check if the aruco XMLs are already present before executing this process?

File: infrastructure/prx_models/models/aruco_mip_36h12/empty_aruco.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<mujoco model="empty_aruco">
+  <compiler angle="radian" />
+  <size njmax="500" nconmax="100"/>
+  <option timestep="0.001" integrator="Euler"/>
```
Comment (aravindsiv) [inline infrastructure/prx_models/models/aruco_mip_36h12/empty_aruco.xml:4]: When `mujoco` includes a file, I am unsure about the behavior of the `option`s, i.e. whether the base XML's options override the includes or not (it shouldn't). Or we can improve the `option`, etc., from this file (from what I can tell, this is only defining the tag bodies, so physics is not very important).

File: world/mujoco_ros/include/mujoco_ros/defs.h [hunk 3]
```diff
@@ -7,8 +7,43 @@
...
 #include <iomanip>
+#include <regex>
+
 #include "GLFW/glfw3.h"
 #include "ros/ros.h"
 #include "mujoco/mujoco.h"
 #include "std_msgs/Empty.h"
 #include "mujoco_ros/SensorDataStamped.h"
+
+#define PRX_DEBUG_PRINT std::cout << __PRETTY_FUNCTION__ << ": " << __LINE__ << std::endl;
```
Comment (aravindsiv) [inline world/mujoco_ros/include/mujoco_ros/defs.h:18]: Nit: Do we want to use `PRX_` prefixs in this repository?

File: world/mujoco_ros/include/mujoco_ros/simulator.hpp [hunk 4]
```diff
@@ -51,6 +63,7 @@ class simulator_t : public std::enable_shared_from_this<simulator_t>
     {
       add_current_state_to_trajectory();
     }
+    std::cout << "timestep: " << m->opt.timestep << std::endl;
```
Comment (aravindsiv) [inline world/mujoco_ros/include/mujoco_ros/simulator.hpp]: Nit: Repeated line

File: world/mujoco_ros/src/mushr_simulation.cpp [hunk 5]
```diff
@@ -34,44 +35,44 @@ int main(int argc, char** argv)
...
+    sim_subscribers.push_back(n.subscribe(root + "/ml4kp_traj", 1000,
+                                          &mj_ros::simulator_visualizer_t::set_trajectory_to_visualize,
+                                          visualizer.get()));
+    sim_subscribers.push_back(n.subscribe(root + "/reset", 1000, &mj_ros::simulator_visualizer_t::reset, visualizer.get()));
   }
 
+  mj_ros::camera_rgb_publisher_t camera_publisher(n, sim, "camera_0");
+  mj_ros::run_simulation(sim, visualizer, 2, camera_publisher);
+
+  // Set the threads
```
Comment (aravindsiv) [inline world/mujoco_ros/src/mushr_simulation.cpp]: Nit: Commented out lines that could be removed for readability

File: infrastructure/interface/src/prx_image_view.cpp
Comment (aravindsiv) [inline infrastructure/interface/src/prx_image_view.cpp:1]: What does this script do that we could not achieve via `rqt_image_view`?

File: world/mujoco_ros/include/mujoco_ros/camera_publisher.hpp [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+    for (int i = 0; i < _sim->m->ncam; ++i)
+    {
+      const std::string name_i{ _sim->m->names + _sim->m->name_camadr[i] };
+      // std::cout << "name_i: " << name_i << std::endl;
+      if (_cam_name == name_i)
+      {
+        _mj_viewport.left = 0;
+        _mj_viewport.bottom = 0;
+        // _mj_viewport.width = _sim->m->cam_resolution[i];
```
Comment (aravindsiv) [inline world/mujoco_ros/include/mujoco_ros/camera_publisher.hpp:59]: I am a fan of this and not hardcoding it, but if there is a specific resolution that we want to use, this is fine.

**Review by aravindsiv — CHANGES_REQUESTED** [review]
Only minor comments -- thanks for the integration!

File: infrastructure/prx_models/models/aruco_mip_36h12/empty_aruco.xml (hunk 2 above)
Comment (garygra) [reply infrastructure/prx_models/models/aruco_mip_36h12/empty_aruco.xml:4]: I was using this to test the aruco tags _in isolation_. This shouldn't be imported into another mj xml...

File: world/mujoco_ros/include/mujoco_ros/defs.h (hunk 3 above)
Comment (garygra) [reply world/mujoco_ros/include/mujoco_ros/defs.h:18]: I copy/pasted this from ML4KP. I don't have a preference.

File: infrastructure/interface/src/prx_image_view.cpp
Comment (garygra) [reply infrastructure/interface/src/prx_image_view.cpp:1]: Nothing really. At some point, I thought the problem was on my `rqt_image_view`, so I used this to test it. Removing it.

File: world/mujoco_ros/include/mujoco_ros/camera_publisher.hpp (hunk 6 above)
Comment (garygra) [reply world/mujoco_ros/include/mujoco_ros/camera_publisher.hpp:59]: Agree with you, but there is a problem: `cam_resolution` is only on mj 3.0 and forward ([23](https://mujoco.readthedocs.io/en/stable/changelog.html#version-3-0-0-october-18-2023)). I'm currently using 2.7.

File: infrastructure/prx_models/models/aruco_mip_36h12/empty_aruco.xml (hunk 2 above)
Comment (aravindsiv) [reply infrastructure/prx_models/models/aruco_mip_36h12/empty_aruco.xml:4]: I checked on 3.0.0 and it seems to happen -- can you check on your version, and remove if unnecessary?

File: world/mujoco_ros/include/mujoco_ros/camera_publisher.hpp (hunk 6 above)
Comment (aravindsiv) [reply world/mujoco_ros/include/mujoco_ros/camera_publisher.hpp:59]: Agreed, could you add a `TODO` for now (We can fix once we upgrade to 3.0)

**Review by aravindsiv — APPROVED** [review]

**Pull request merged**